### PR TITLE
fix(watermark): align watermark for idle input

### PR DESCRIPTION
**Pull request by yuhao-su**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- When a `WatermarkFilter` has an idle input, it aligns with the global max watermark.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by st1page — APPROVED** [review]

File: src/stream/src/executor/watermark_filter.rs [hunk 1]
```diff
@@ -229,6 +234,19 @@ impl<S: StateStore> WatermarkFilterExecutor<S> {
                             table.insert(row);
                         }
                         table.commit(barrier.epoch).await?;
+
+                        if idle_input {
```
Comment (BugenZhao) [inline src/stream/src/executor/watermark_filter.rs:238]: I'm unsure if it's reasonable to use a single checkpoint as the threshold for determining an idle source, considering that remote reads are involved. Let's wait and see.

**Review by BugenZhao — COMMENTED** [review]
LGTM

File: src/stream/src/executor/watermark_filter.rs [hunk 2]
```diff
@@ -229,6 +234,19 @@ impl<S: StateStore> WatermarkFilterExecutor<S> {
...
+                            // Align watermark
+                            let global_max_watermark =
+                                Self::get_global_max_watermark(&table, watermark_type.clone()).await?;
```
[the remark this answers is not in the document]
Comment (yuhao-su) [reply src/stream/src/executor/watermark_filter.rs:241]: 1. Watermark can't be None.
2. Min watermark will make downstream fail to use any meaningful watermark todo clean/eowc

[the remark this answers is not in the document]
Comment (liurenjie1024) [reply src/stream/src/executor/watermark_filter.rs]: We can change watermark to be None. None means that the actor has no watermark, and watermark mergers can ignore this. Anyway it's just another way to do it.

[the remark this answers is not in the document]
Comment (yuhao-su) [reply src/stream/src/executor/watermark_filter.rs]: No. If the partition gives a low watermark after None, it will requires merger to do filter work

**Pull request merged**